generate-rs/src/parser.rs

```diff
@@ -0,0 +1,593 @@
+    //                 typ = field[0]
+    //             else:
+    //                 typ = field
+    //             return (optionalize('HashMap<String,' + typ + '>', optional), schema.get('description', '')), structs
+
+    //     if schema['type'] == 'array':
+    //         typ, substructs = parse_schema_types(name, schema['items'], optional=False, parents=parents + [name])
+    //         if type(typ) is tuple:
+    //             typ = typ[0]
+    //         return (optionalize('Vec<' + typ + '>', optional), schema.get('description', '')), structs + substructs
+
+    //     if schema['type'] == 'string':
+
+    //         def build(intt, typ='String'):
+    //             return (optionalize(typ, optional), intt + ': ' + schema.get('description', '')), structs
+
+    //         if 'format' in schema:
+    //             if schema['format'] == 'int64':
+    //                 return build('i64')
+    //             if schema['format'] == 'int32':
+    //                 return build('i32')
+    //             if schema['format'] == 'uint64':
+    //                 return build('u64')
+    //             if schema['format'] == 'uint32':
+    //                 return build('u32')
+    //             if schema['format'] == 'double':
+    //                 return build('f64')
+    //             if schema['format'] == 'float':
+    //                 return build('f32')
+    //             if schema['format'] == 'date-time':
+    //                 return build('DateTime', typ='DateTime<Utc>')
+    //         return (optionalize('String', optional), schema.get('description', '')), structs
+
+    //     if schema['type'] == 'boolean':
+    //         return (optionalize('bool', optional), schema.get('description', '')), structs
+
+    //     if schema['type'] in ('number', 'integer'):
+
+    //         def build(intt):
+    //             return (optionalize(intt, optional), schema.get('description', '')), structs
+
+    //         if schema['format'] == 'float':
+    //             return build('f32')
+    //         if schema['format'] == 'double':
+    //             return build('f64')
+    //         if schema['format'] == 'int32':
+    //             return build('i32')
+    //         if schema['format'] == 'int64':
+    //             return build('i64')
+    //         if schema['format'] == 'uint32':
+    //             return build('u32')
+    //         if schema['format'] == 'uint64':
+    //             return build('u64')
+
+    //     if schema['type'] == 'any':
+    //         return (optionalize('String', optional), 'ANY data: ' + schema.get('description', '')), structs
+
+    //     raise Exception('unimplemented schema type!', name, schema)
+    // except KeyError as e:
+    //     print('KeyError while processing:', name, schema)
+    //     raise e
+}
+
+fn generate_params_structs(resources: Vec<String>, super_name: Option<String>, global_params: Option<String>) -> Vec<String> {
```

Comment:
this may benefit from being packed into a `ParametersGenerator`, which keeps state such as the global parameters and so on.

Then the whole parameter types generating business could be a subprogram.

Reply:
Agreed.